### PR TITLE
PullRequest for Issue1318: Build convenience function for AMMainWindow

### PR DESCRIPTION
**Pull request by XuetaoLiu-CLS**
@dleacock 
I created this PR, so that it is easier to review

https://github.com/acquaman/acquaman/issues/1318

### CODE REVIEW

**Comment by dleacock** [comment]
@dretrex  Here are the only changes I've made.  Very similar to what I had last week, minus the groupBox. I used a widget instead. Unsure what is causing the grey overlap over the top frame.  I used setContentMargins (0, -40, 0 ,0) to push it as high as I could before the grey covers it up. 

cddff21968565375e4245b8b457f89f68f385e99

**Comment by dretrex** [comment]
Hey @dleacock, although it took me probably 3 hours of playing around, the issue is that there is an empty QLabel called disabled warning.  That is taking up the space at the top.  I'd say you could probably remove it (from both constructors) unless @davidChevrier has any objections.  Then you could use something like,

```
topFrame_ = new AMTopFrame(frameName);
topFrame_->setIcon(QIcon(":/utilities-system-monitor.png"));

QHBoxLayout *HLayout = new QHBoxLayout;
HLayout->addStretch();
HLayout->addWidget(view_);
HLayout->addStretch();

layout_->insertStretch(0);
layout_->insertLayout(0, HLayout);
layout_->insertStretch(0);
layout_->insertWidget(0, topFrame_);
```

**Comment by dleacock** [comment]
Added:

* Additional constructor for AMScanConfigurationViewHolder that takes the view, the name of the frame and adds it to the layout with an optional squeeze. 

58962f9

File: source/ui/BioXAS/BioXASSideXASScanConfigurationView.cpp [hunk 1]
```diff
@@ -40,8 +40,8 @@ BioXASSideXASScanConfigurationView::BioXASSideXASScanConfigurationView(BioXASSid
 {
 	configuration_ = configuration;
 
-	topFrame_ = new AMTopFrame("Configure an XAS Scan");
-	topFrame_->setIcon(QIcon(":/utilities-system-monitor.png"));
+	//topFrame_ = new AMTopFrame("Configure an XAS Scan");
```
Comment (dretrex) [inline source/ui/BioXAS/BioXASSideXASScanConfigurationView.cpp]: Have you been able to run these applications and see if they are working as intended?  If so, you should delete the commented out lines.

**Comment by dretrex** [comment]
Once you remove those commented lines, this looks good to me.  I'll merge it in, unless there is more you'd like to do first?

**Comment by dleacock** [comment]
Im just working on the BioXASSide changes right now. I'll have a cleaner
commit shortly.

**Comment by dretrex** [comment]
Okay.  I'm going to add the "Work in progress" label to this.  Once it's ready, flag it with the "Ready for Review" label and ask for the final review.

File: source/ui/SXRMB/SXRMBEXAFSScanConfigurationView.cpp [hunk 2]
```diff
@@ -35,7 +35,7 @@ SXRMBEXAFSScanConfigurationView::SXRMBEXAFSScanConfigurationView(SXRMBEXAFSScanC
 
 	configuration_ = configuration;
 
-	topFrame_ = new AMTopFrame("Configure an XAS Scan");
+	//topFrame_ = new AMTopFrame("Configure an XAS Scan");
```
Comment (dretrex) [inline source/ui/SXRMB/SXRMBEXAFSScanConfigurationView.cpp]: There is still a commented out line here.

**Comment by dretrex** [comment]
Hey @dleacock, could you go into the AMGenericScanConfigurationView and remove the top frame from there as well?  Thanks.

**Pull request merged**